File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        count = 0
        i = 0
        l = len(intervals)-1
        intervals.sort()
        while count < l:
            # If the two intervals are overlapping,
            if intervals[i][1] >= intervals[i+1][0]: 
                maxY = max(intervals[i][1], intervals[i+1][1])
                intervals[i][1] = maxY
                intervals.pop(i+1)
            else:
                i+=1
            count+=1
        return intervals

File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> res;
        
        // Sort
        sort(intervals.begin(), intervals.end());
        
        for (auto i : intervals){
            if (res.empty()){
                res.push_back(i);
            } else {
                if(res[res.size()-1][1] >= i[0]){
                    vector<int> newInterval = {res[res.size()-1][0], max(res[res.size()-1][1], i[1])};
                    res[res.size()-1] = newInterval;
                } else {
                    res.push_back(i);
                }
            }
        }
        
        
        return res;
    }
};